config.ru

```diff
@@ -4,16 +4,41 @@ require 'routemaster/receiver'
 require 'dotenv'
 require 'sinatra'
 require 'pry'
+require 'data_sink_client'
+require 'concurrent/fixed_thread_pool'
 
 Dotenv.load!
 
+SUBSCRIBER_NAME = 'routemaster-blackhole'
+
 class Handler
   def on_events(events)
     events.each do |event|
       event.keys.each { |k| event[k.to_sym] = event.delete(k) }
       $stderr.write("%<topic>-12s %<type>-12s %<url>s %<t>s\n" % event)
       $stderr.flush
     end
+    send_to_data_sink(events)
+  end
+
+  private
+
+  def send_to_data_sink(events)
+    return unless data_sink_enabled?
+    events.map do |event|
+      pool = Concurrent::FixedThreadPool.new(Concurrent.processor_count)
+      pool.post do
+        event = event.merge(
+          subscriber: SUBSCRIBER_NAME,
+          received_at: (Time.now.utc.to_f * 1e3).to_i
+        )
+        DataSinkClient.instance.post(event)
+      end
+    end.map(&:join)
```

Comment:
I think this makes `Concurrent.processor_count` threads per event. 😃 

I think you'll also need to change the `.map(&:join)`. It seems `FixedThreadPool` has a `#wait_for_termination`.

Reply:
I wanted to try this locally first and pushed to remote by mistake. Forget everyone gets an email 😅

Reply:
Aha, I thought that might be the case. :) I wouldn't have mentioned anything had the PR not still been 'approved'.